components/Loader.tsx

```diff
@@ -0,0 +1,23 @@
+import { LoadingSkeleton } from "components";
+import type { ComponentProps, ReactNode } from "react";
+type LoadingSkeletonProps = ComponentProps<typeof LoadingSkeleton>;
+
+type Props = LoadingSkeletonProps & {
+  isLoading: boolean;
+  children: ReactNode;
+  isError?: boolean;
+  error?: ReactNode;
+};
+export function Loader({
+  isLoading,
+  isError,
+  children,
+  error,
+  ...delegated
+}: Props) {
+  if (isError && !!error) return <>{error}</>;
+
+  if (isLoading) return <LoadingSkeleton {...delegated} />;
+
+  return <>{children}</>;
+}
```

Comment:
in your first iteration you had a way to set a default state, which was the state that showed when no wallet was connected, now it just shows loading if wallet is not connected. i know we discussed this but dont remember if this changed behavior was intentional